### PR TITLE
3 packages from roddyyaga/ppx_rapper at 3.0.0

### PR DESCRIPTION
**Pull request by roddyyaga**
This pull-request concerns:
-`ppx_rapper.3.0.0`: Syntax extension for Caqti/PostgreSQL queries
-`ppx_rapper_async.3.0.0`: Async support for ppx_rapper
-`ppx_rapper_lwt.3.0.0`: Lwt support for ppx_rapper

---
* Homepage: https://github.com/roddyyaga/ppx_rapper
* Source repo: git+https://github.com/roddyyaga/ppx_rapper.git
* Bug tracker: https://github.com/roddyyaga/ppx_rapper/issues

---
:camel: Pull-request generated by opam-publish v2.0.2

### CODE REVIEW

**Comment by kit-ty-kate** [comment]
Your tests fail with:
```
#=== ERROR while compiling ppx_rapper.3.0.0 ===================================#
# context              2.0.7 | linux/x86_64 | ocaml-base-compiler.4.07.1 | pinned(https://github.com/roddyyaga/ppx_rapper/archive/3.0.0.tar.gz)
# path                 ~/.opam/4.07/.opam-switch/build/ppx_rapper.3.0.0
# command              ~/.opam/4.07/bin/dune build -p ppx_rapper -j 31 @install @runtest
# exit-code            1
# env-file             ~/.opam/log/ppx_rapper-19-8b9749.env
# output-file          ~/.opam/log/ppx_rapper-19-8b9749.out
### output ###
# File "test/runtime/dune", line 3, characters 17-31:
# 3 |  (libraries base ppx_rapper_lwt caqti-type-calendar)
#                      ^^^^^^^^^^^^^^
# Error: Library "ppx_rapper_lwt" not found.
# Hint: try:
#   dune external-lib-deps --missing -p ppx_rapper -j 31 @install @runtest
```
I've sent a fix to https://github.com/roddyyaga/ppx_rapper/pull/11

File: packages/ppx_rapper/ppx_rapper.3.0.0/opam [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+doc: "https://github.com/roddyyaga/ppx_rapper"
+bug-reports: "https://github.com/roddyyaga/ppx_rapper/issues"
+depends: [
+  "ocaml" {>= "4.07"}
+  "dune" {>= "2.0.1"}
+  "pg_query"
+  "ppxlib"
+  "base"
+  "caqti"
+  "caqti-type-calendar" {with-test}
```
Comment (kit-ty-kate) [inline packages/ppx_rapper/ppx_rapper.3.0.0/opam:16]: This should be in `ppx_rapper_async` and `ppx_rapper_lwt` instead. It is not used by this package

File: packages/ppx_rapper/ppx_rapper.3.0.0/opam [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+license: "MIT"
+homepage: "https://github.com/roddyyaga/ppx_rapper"
+doc: "https://github.com/roddyyaga/ppx_rapper"
+bug-reports: "https://github.com/roddyyaga/ppx_rapper/issues"
+depends: [
+  "ocaml" {>= "4.07"}
+  "dune" {>= "2.0.1"}
+  "pg_query"
+  "ppxlib"
+  "base"
```
Comment (kit-ty-kate) [inline packages/ppx_rapper/ppx_rapper.3.0.0/opam:14]: These two should most likely have a lower-bound constraint.
I would suspect something like
```
  "ppxlib" {>= "0.14.0"}
  "base" {>= "v0.9.0"}
```
or something like that, at the very least

File: packages/ppx_rapper_lwt/ppx_rapper_lwt.3.0.0/opam [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+maintainer: ["Roddy MacSween <github@roddymacsween.co.uk>"]
+authors: ["Roddy MacSween <github@roddymacsween.co.uk>"]
+license: "MIT"
+homepage: "https://github.com/roddyyaga/ppx_rapper"
+doc: "https://github.com/roddyyaga/ppx_rapper"
+bug-reports: "https://github.com/roddyyaga/ppx_rapper/issues"
+depends: [
+  "ocaml" {>= "4.07"}
+  "dune" {>= "2.0.1"}
+  "ppx_rapper"
```
Comment (kit-ty-kate) [inline packages/ppx_rapper_lwt/ppx_rapper_lwt.3.0.0/opam:12]: If you're planning to release the three packages at the same time, could you sync up the version number as follow:
```
  "ppx_rapper" {= version}
```

File: packages/ppx_rapper_async/ppx_rapper_async.3.0.0/opam [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+homepage: "https://github.com/roddyyaga/ppx_rapper"
+doc: "https://github.com/roddyyaga/ppx_rapper"
+bug-reports: "https://github.com/roddyyaga/ppx_rapper/issues"
+depends: [
+  "ocaml" {>= "4.07"}
+  "dune" {>= "2.0.1"}
+  "ppx_rapper"
+  "caqti-async"
+  "caqti"
+  "async"
```
Comment (kit-ty-kate) [inline packages/ppx_rapper_async/ppx_rapper_async.3.0.0/opam:15]: same here (though I'm not sure how the caqti API changed over time as I'm not familiar with it)

**Comment by roddyyaga** [comment]
Thanks, I've fixed those things.

**Comment by kit-ty-kate** [comment]
This error (https://github.com/ocaml/opam-repository/pull/18106#issuecomment-774709606) is still there. My fix did not make it to the tag. Could upstream my fixes (just so i don't have to do them again) and re-tag 3.0.0 with https://github.com/roddyyaga/ppx_rapper/pull/11 included ?

**Comment by kit-ty-kate** [comment]
Sorry but it seems you overwrote my fixes:
```
#=== ERROR while compiling ppx_rapper.3.0.0 ===================================#
# context              2.0.8 | linux/x86_64 | ocaml-base-compiler.4.07.1 | pinned(https://github.com/roddyyaga/ppx_rapper/archive/3.0.0.tar.gz)
# path                 ~/.opam/4.07/.opam-switch/build/ppx_rapper.3.0.0
# command              ~/.opam/4.07/bin/dune build -p ppx_rapper -j 31 @install
# exit-code            1
# env-file             ~/.opam/log/ppx_rapper-15-45e01a.env
# output-file          ~/.opam/log/ppx_rapper-15-45e01a.out
### output ###
# File "lib-runtime/dune", line 4, characters 12-17:
# 4 |  (libraries caqti))
#                 ^^^^^
# Error: Library "caqti" not found.
# Hint: try:
#   dune external-lib-deps --missing -p ppx_rapper -j 31 @install
```

**Comment by kit-ty-kate** [comment]
Thanks a lot!

**Pull request merged**